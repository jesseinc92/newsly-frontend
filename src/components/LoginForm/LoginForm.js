import React, { useContext } from "react";
import { Link, Navigate, useNavigate } from 'react-router-dom';
import useFormChange from "../../hooks/useFormChange";
import UserContext from "../../UserContext";
import NewslyAPI from "../../api";
import Button from '../Button/Button';

const LoginForm = () => {
  const { user, setUser } = useContext(UserContext);
  const redirect = useNavigate();

  const [formData, handleChange] = useFormChange({
      username: '',
      password: ''
  });

  const handleLogin = async e => {
    e.preventDefault();

    try {
      await NewslyAPI.loginUser(formData);

      // API call to get user
      const loggedUser = await NewslyAPI.getUser(formData.username);
      localStorage.setItem('user', JSON.stringify(loggedUser));
      setUser(loggedUser);

      // Once form is submitted redirect back to homepage.
      redirect('/');
    } catch(err) {
      alert(err);
    }
  }

  if (!user) {
    return (
      <main>
        <p>Don't yet have an account? <Link to='/user/register'>Sign Up</Link> here.</p>
        <form onSubmit={handleLogin}>
          <div className='input-group'>
            <label htmlFor='username'>Username</label>
            <input 
              id='username' 
              name='username' 
              type='text' 
              value={formData.username} 
              onChange={handleChange} 
              autoFocus
            />
          </div>
          <div className='input-group'>
            <label htmlFor='password'>Password</label>
            <input 
              id='password' 
              name='password' 
              type='password' 
              value={formData.password} 
              onChange={handleChange} 
            />
          </div>
          <Button text='Login' />
        </form>
      </main>
    )
  } else {
    return <Navigate to='/' />
  }
}

export default LoginForm;