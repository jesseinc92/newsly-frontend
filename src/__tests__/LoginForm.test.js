import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import LoginForm from '../components/LoginForm/LoginForm';
import renderWithContext from '../helpers/testContext';

it('renders the component', function() {
  renderWithContext(<MemoryRouter><LoginForm /></MemoryRouter>, null);
});

it('matches the snapshot', function() {
  const { asFragment } = renderWithContext(<MemoryRouter><LoginForm /></MemoryRouter>, null)
  expect(asFragment()).toMatchSnapshot();
})

