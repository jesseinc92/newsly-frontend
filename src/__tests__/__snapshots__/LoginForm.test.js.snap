// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="LoginForm"
  >
    <p>
      Don't have an account? 
      <a
        class="underlined-link"
        href="/user/register"
      >
        Sign Up
      </a>
       here.
    </p>
    <form>
      <div
        class="input-group"
      >
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          name="username"
          type="text"
          value=""
        />
      </div>
      <div
        class="input-group"
      >
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </div>
      <input
        class="LoginForm-submit"
        type="submit"
        value="Login"
      />
    </form>
  </main>
</DocumentFragment>
`;
